:sparkles: Home뷰 기본 틀 구현 (#9, #10, #11, #12)

cws653 opened the pull request:
# 개요
- 현재 디자인 미확정 및 Component 미확정에 따라 향후 디테일하게 수정 예정
- Home, 친구랭킹 리스트, 문제리스트, 문제만들기 페이지 구현
- 🔗  이슈링크 : #9 #10 #11 #12 

# 작업 내용
<!-- 작업한 코드/UI에 대한 설명을 작성합니다. -->
1. Home 뷰 구현 : 하이라키뷰와 연결 완료
2. 친구랭킹 리스트 디테일뷰 구현
3. 문제리스트 구현
4. 문제만들기 페이지 구현|

cws653 committed:
:sparkles: Home뷰 기본 틀 구현 (#9, #10, #11, #12)

# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

################
# 세부 내용을 아랫줄에 작성
- 현재 디자인 미확정 및 Component 미확정에 따라 향후 디테일하게 수정 예정
- Home, 친구랭킹 리스트, 문제리스트, 문제만들기 페이지 구현

################
# 추가적인 세부 내용을 아랫줄에 작성

################
# :tada: = 프로젝트 시작
# :sparkles: 
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시
# :rocket: = release 배포 시
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우)
# :memo: = 문서 작성 시
# :art: = 디자인 시스템 관련 내용 (Color / Asset)
# :hammer: = 테스트 코드 작성
################

dittoo8 commented on Projects/Home/Targets/HomeUI/Sources/View/Home.swift:
```diff
@@ -0,0 +1,156 @@
+        }
+    }
+    
+    var questionExpainView: some View {
```
ExpainView가 뭔가욤 ?? 오타일까요 ??

cws653 replied:
옙 오타입니다. ExplainView를 적으려고 했었습니다. 오타 수정하도록 하겠습니다!

szzang-kr commented on Projects/Home/Targets/HomeUI/Sources/View/FriendsList.swift:
```diff
@@ -0,0 +1,33 @@
+    }
+}
+
+//struct FriendsList_Previews: PreviewProvider {
+//    static var previews: some View {
+//        FriendsList(friend: friendsRankingSample[0])
+//    }
+//}
```
사용하지 않는 코드는 삭제해주세요 :)

cws653 replied:
옙 사용하지 않는 코드는 삭제하도록 하겠습니다!

szzang-kr commented on Projects/Home/Targets/HomeUI/Sources/View/Home.swift:
```diff
@@ -0,0 +1,156 @@
+                }
+                .padding()
+            }
+//            .navigationTitle("HOME")
```
사용하지 않는 코드는 삭제해주세요 :)

szzang-kr commented on Projects/Home/Targets/HomeUI/Sources/View/QuestionDetail.swift:
```diff
@@ -0,0 +1,68 @@
+    }
+}
+
+//struct QuestionDetailView_Previews: PreviewProvider {
+//    static var previews: some View {
+//        QuestionDetail(questionGroup: )
+//    }
+//}
```
사용하지 않는 코드는 삭제해주세요 :)

cws653 replied:
옙 사용하지 않는 코드는 삭제하도록 하겠습니다!

szzang-kr commented on Projects/Home/Targets/HomeUI/Sources/ViewModel/HomeViewModel.swift:
```diff
@@ -0,0 +1,26 @@
+    public init() {
+        self.questionGroups = questionsSamlple
+        self.friendsRank = friendsRankSample
+//        self.friendsRank = []
```
사용하지 않는 코드는 삭제해주세요 :)

cws653 replied:
옙 사용하지 않는 코드는 삭제하도록 하겠습니다!

szzang-kr commented on Projects/Home/Targets/HomeUI/Sources/View/QuestionGroupList.swift:
```diff
@@ -0,0 +1,76 @@
+    }
+}
+
+//struct QuestionsList_Previews: PreviewProvider {
+//    static var previews: some View {
+//        QuestionsList(question: questionsSamlple[0])
+//    }
+//}
```
사용하지 않는 코드는 삭제해주세요 :)

szzang-kr commented on Projects/Home/Targets/HomeUI/Sources/View/QuestionDetail.swift:
```diff
@@ -0,0 +1,68 @@
+    }
+}
+
+struct questionsList: View {
```
해당 파일에서만 사용되는 View라면 private을 사용해서 외부의 불필요한 접근을 막을 수 있습니다

cws653 replied:
옙 알겠습니다. 접근제한자를 통해서 외부에서 접근이 가능한 것인지 아닌지 명확하게 구분할 수 있도록 처리해놓겠습니다!

szzang-kr commented on Projects/Home/Targets/HomeUI/Sources/View/QuestionDetailRow.swift:
```diff
@@ -0,0 +1,24 @@
+    }
+}
+
+//struct QuestionDetailRow_Previews: PreviewProvider {
+//    static var previews: some View {
+//        QuestionDetailRow()
+//    }
+//}
```
사용하지 않는 코드는 삭제해주세요 :)

cws653 replied:
옙 사용하지 않는 코드는 삭제하도록 하겠습니다!

cws653 committed:
# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력

# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

################
# 세부 내용을 아랫줄에 작성

################
# 추가적인 세부 내용을 아랫줄에 작성

################
# :tada: = 프로젝트 시작
# :sparkles: = create 파일 생성 시
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시
# :rocket: = release 배포 시
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우)
# :memo: = 문서 작성 시
# :art: = 디자인 시스템 관련 내용 (Color / Asset)
# :hammer: = 테스트 코드 작성
################

cws653 committed:
💡 pullrequest 내용 반영

cws653 committed:
Merge remote-tracking branch 'origin/feature/ws/home' into feature/ws/home
